### PR TITLE
[AArch64][GISel] Support neon.abs intrinsic for vector types

### PR DESCRIPTION
**Pull request by madhur13490**
This patch adds support in instruction selection
aarch64.neon.abs intrinsic for vector types.

### CODE REVIEW

**Comment by davemgreen** [comment]
If we don't have tablegen patterns for these intrinsics it is probably best to do the same as ISel and convert them into a G_ABS, which should be selected naturally.

(Considering the intrinsic, which as far as I understand always works like an abs instruction, it would be nice if aarch64_neon_abs was not generated at-all and it used llvm.abs everywhere instead, but that is a much larger change that is outside the scope of getting them working with gisel).

I believe there are other intrinsics that should work the same way. Perhaps a combine could be added to replace the intrinsic with abs, early enough that the other optimizations we would ideally want can happen. The good thing about doing it earlier is it shouldn't have to be as careful about all the different types, it can just convert the intrinsic.

File: llvm/lib/Target/AArch64/GISel/AArch64InstructionSelector.cpp [hunk 1]
```diff
@@ -6644,6 +6644,34 @@ bool AArch64InstructionSelector::selectIntrinsic(MachineInstr &I,
   switch (IntrinID) {
   default:
     break;
+  case Intrinsic::aarch64_neon_abs: {
+    Register DstReg = I.getOperand(0).getReg();
+    Register SrcReg = I.getOperand(2).getReg();
+    auto SrcRegType = MRI.getType(SrcReg);
+    unsigned Opc = 0;
+
+    if (SrcRegType == LLT::fixed_vector(8, 8)) {
```
Comment (arsenm) [inline llvm/lib/Target/AArch64/GISel/AArch64InstructionSelector.cpp:6653]: This looks like it should be easily handled with patterns

**Comment by madhur13490** [comment]
@arsenm @davemgreen 
Please have a look. I made the changes as per @davemgreen's suggestion.

**Review by davemgreen — APPROVED** [review]
Thanks, LGTM

**Pull request merged**